service/src/main/java/bio/terra/cbas/runsets/inputs/InputGenerator.java

```diff
@@ -34,7 +36,12 @@ public static Map<String, Object> buildInputs(
       } else {
         String attributeName =
             ((ParameterDefinitionRecordLookup) param.getSource()).getRecordAttribute();
-        parameterValue = record.getAttributes().get(attributeName);
+
+        if (!((Map<String, Object>) recordResponse.getAttributes()).containsKey(attributeName)) {
```

Comment:
The short answer: A HashMap.

Reply:
The short answer: A HashMap.

Reply:
It's casting a lowest-common-denominator type (`recordResponse.getAttributes()` is just an `Object`) to something more specific, so that it has a `containsKey` method to call